src/vm-memory/src/mmap.rs

```diff
@@ -51,9 +77,39 @@ impl GuestRegionMmap {
         Ok(GuestRegionMmap {
             mapping,
             guest_base,
+            dirty_bitmap: None,
+        })
+    }
+
+    /// Create a new memory-mapped memory region for the guest's physical memory.
+    /// This region will own a dedicated bitmap to handle dirty page tracking.
+    pub fn new_with_bitmap(
+        mapping: MmapRegion,
+        guest_base: GuestAddress,
+        bitmap: Bitmap,
+    ) -> result::Result<Self, Error> {
+        if guest_base.0.checked_add(mapping.len() as u64).is_none() {
+            return Err(Error::InvalidGuestRegion);
+        }
+        Ok(GuestRegionMmap {
+            mapping,
+            guest_base,
+            dirty_bitmap: Some(bitmap),
```

Comment:
Shouldn't we build the bitmap here ? Since we know all the details about the region. What's the advantage of accepting the bitmap as a parameter ?